textile/features.textile

```diff
@@ -1406,11 +1412,139 @@ h4. PresenceMessage
+** @(SM2d)@ @extras@ JSON-encodable object, used to contain any arbitrary key value pairs which may also contain other primitive JSON types, JSON-encodable objects or JSON-encodable arrays. The @extras@ field is provided to contain message metadata and/or ancillary payloads in support of specific functionality. For 3.1 no specific functionality is specified for @extras@ in objects messages. Unless otherwise specified, the client library should not attempt to do any filtering or validation of the @extras@ field itself, but should treat it opaquely, encoding it and passing it to realtime unaltered
+** @(SM2e)@ @timestamp@ time in milliseconds since epoch. If an objects message received from Ably does not contain a @timestamp@, it should be set to the @timestamp@ of the encapsulating @ProtocolMessage@
+** @(SM2f)@ @operation@ @StateOperation@ object
+** @(SM2g)@ @object@ @StateObject@ object
+** @(SM2h)@ @serial@ string - the origin timeserial of this objects message
+** @(SM2i)@ @siteCode@ string - the site code corresponding to this objects message's timeserial in the @serial@ property
+* @(SM3)@ The size of the @StateMessage@ for "TO3l8":#TO3l8 is calculated as follows:
+** @(SM3a)@ The size is the sum of the sizes of the @clientId@, @operation@, @object@, and @extras@ properties
+** @(SM3b)@ The size of the @operation@ property is calculated per "SOP4":#SOP4
+** @(SM3c)@ The size of the @object@ property is calculated per "SOB3":#SOB3
+** @(SM3d)@ The size of the @extras@ property is the string length of its JSON representation
+** @(SM3e)@ The size of a @null@ or omitted property is zero
+
+h4. StateOperation
+
+* @(SOP1)@ A @StateOperation@ describes an operation to be applied to an object on a channel
+* @(SOP2)@ @StateOperation@ @Action@ enum has the following values in order from zero: @MAP_CREATE@, @MAP_SET@, @MAP_REMOVE@, @COUNTER_CREATE@, @COUNTER_INC@, @OBJECT_DELETE@
+* @(SOP3)@ The attributes available in a @StateOperation@ are:
+** @(SOP3a)@ @action@ enum - defines the operation to be applied to the object
+** @(SOP3b)@ @objectId@ string - the object ID of the object on a channel to which the operation should be applied
+** @(SOP3c)@ @mapOp@ @MapOp@ object - the payload for the operation if it is an operation on a @Map@ object type
+** @(SOP3d)@ @counterOp@ @CounterOp@ object - the payload for the operation if it is an operation on a @Counter@ object type
+** @(SOP3e)@ @map@ @Map@ object - the payload for the operation if the operation is @MAP_CREATE@. Defines the initial value for the @Map@ object
+** @(SOP3f)@ @counter@ @Counter@ object - the payload for the operation if the operation is @COUNTER_CREATE@. Defines the initial value for the @Counter@ object
+** @(SOP3g)@ @nonce@ string - the nonce, must be present on objects create operations sent to the server
+** @(SOP3h)@ @initialValue@ binary - the initial value bytes for the object
```

Comment:
Do we have a spec point that defines how to encode and decode a binary property of a `ProtocolMessage`?